dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -1276,49 +1277,50 @@ void dpnp_rng_vonmises_large_kappa_c(void* result, const _DataType mu, const _Da
     Vvec = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(size * sizeof(_DataType)));
-
-    for (size_t n = 0; n < size;)
+    n = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(sizeof(size_t)));
+    for (*n = 0; *n < size;)
     {
-        size_t diff_size = size - n;
+        size_t diff_size = size - *n;
         mkl_rng::uniform<_DataType> uniform_distribution_u(d_zero, 0.5 * M_PI);
-        auto event_out = mkl_rng::generate(uniform_distribution_u, DPNP_RNG_ENGINE, diff_size, Uvec);
-        event_out.wait();
-        // TODO
-        // use deps case
+        auto uniform_distr_u_event = mkl_rng::generate(uniform_distribution_u, DPNP_RNG_ENGINE, diff_size, Uvec);
         mkl_rng::uniform<_DataType> uniform_distribution_v(d_zero, d_one);
-        event_out = mkl_rng::generate(uniform_distribution_v, DPNP_RNG_ENGINE, diff_size, Vvec);
-        event_out.wait();
+        auto uniform_distr_v_event = mkl_rng::generate(uniform_distribution_v, DPNP_RNG_ENGINE, diff_size, Vvec);
 
-        // TODO
-        // kernel
-        for (size_t i = 0; i < diff_size; i++)
-        {
-            _DataType sn, cn, sn2, cn2;
-            _DataType neg_W_minus_one, V, Y;
+        cl::sycl::range<1> diff_gws(diff_size);
+        auto paral_kernel_some = [&](cl::sycl::handler& cgh) {
+            cgh.depends_on({uniform_distr_u_event, uniform_distr_v_event});
+            cgh.parallel_for(diff_gws, [=](cl::sycl::id<1> global_id) {
+                size_t i = global_id[0];
 
-            sn = sin(Uvec[i]);
-            cn = cos(Uvec[i]);
-            V = Vvec[i];
-            sn2 = sn * sn;
-            cn2 = cn * cn;
+                _DataType sn, cn, sn2, cn2;
+                _DataType neg_W_minus_one, V, Y;
 
-            neg_W_minus_one = s_minus_one * sn2 / (0.5 * s_minus_one + cn2);
-            Y = kappa * (s_minus_one + neg_W_minus_one);
+                sn = cl::sycl::sin(Uvec[i]);
+                cn = cl::sycl::cos(Uvec[i]);
+                V = Vvec[i];
+                sn2 = sn * sn;
+                cn2 = cn * cn;
 
-            if ((Y * (2 - Y) >= V) || (log(Y / V) + 1 >= Y))
-            {
-                Y = neg_W_minus_one * (2 - neg_W_minus_one);
-                if (Y < 0)
-                    Y = 0.0;
-                else if (Y > 1.0)
-                    Y = 1.0;
+                neg_W_minus_one = s_minus_one * sn2 / (0.5 * s_minus_one + cn2);
+                Y = kappa * (s_minus_one + neg_W_minus_one);
 
-                result1[n++] = asin(sqrt(Y));
-            }
-        }
+                if ((Y * (2 - Y) >= V) || (cl::sycl::log(Y / V) + 1 >= Y))
+                {
+                    Y = neg_W_minus_one * (2 - neg_W_minus_one);
+                    if (Y < 0)
+                        Y = 0.0;
+                    else if (Y > 1.0)
+                        Y = 1.0;
+                    *n = *n + 1;
+                    result1[*n] = cl::sycl::asin(cl::sycl::sqrt(Y));
```

Comment:
Looks like here we are getting race condition, that is why we are getting wrong results. To prevent it we should calculate `n` (index of result) from `i`.